elastic/kibana
# Removes legacy infra plugin and moves saved objects registration to NP

Closes #57999
Closes #61773

## Summary

This PR moves our logs/metrics saved object registration to the new platform and removes the legacy infra plugin as it's no longer needed 🎉 🎉 🎉 .

In doing this, I've also made all 3 of our saved objects importable and exportable.

To test this:
- visit the logs or metrics section to verify that you can modify the settings in each
- visit the metrics explorer and save a view
- visit the metrics inventory and save a view
- navigate to the management section and look in the saved objects list, you should see 3 saved objects saved there that are now all exportable

## Outstanding questions

These have been asked in Slack in #kibana-platform, but still need to be addressed:

- [x] The migration guide here: https://github.com/elastic/kibana/blob/master/src/core/MIGRATION_EXAMPLES.md#saved-objects-types
says there is a value called `namespaceAgnostic` that used to be `isNamespaceAgnostic`, but the `SavedObjectType` TS type I'm using says the field is `namespaceType` and that it's required. And I don't know what it is or does (I also don't know what hidden is or does, which is another required field, but I guessed on that one)
- [ ] I'd also like to understand what all of the "management" and "migration" field options do ... what's the best way to discover these things?
- [ ] also, it looks like the `SavedObjectsType` from `src/core/server` isn't a generic, so is it not possible to type the mappings?
